### ISSUE TITLE
Gin index created each run

### ISSUE THREAD

**Issue by vadimline** [JasperFx/marten#702]
On each application run Marten re-create index, is it correct?

```
public class Clip
{
  public int Id { get; set; }
  public string Name { get; set; }
}

var store = DocumentStore.For(_ =>
{
   _.Connection(connectionString);
   _.AutoCreateSchemaObjects = AutoCreate.CreateOrUpdate;
   _.Schema.For<Clip>()
   .HiloSettings(new Marten.Schema.Identity.Sequences.HiloSettings()
   {
    MaxLo = 10
   })
   .GinIndexJsonData()
    .SoftDeleted();
});
```

Schema already exists in Pg and when I run store.Schema.AssertDatabaseMatchesConfiguration()
I have next result below each time
`-- SchemaDiff for Library.Clip
drop index public.mt_doc_clip_idx_data;
CREATE INDEX mt_doc_clip_idx_data ON public.mt_doc_clip USING gin ("data" jsonb_path_ops);;`

Maybe last semi column an issue?

**Comment by vadimline** [comment]
Did some debug and found the the problem

IndexDefinition.Matches work incorrect and particularly next lines

```
            _columns.Each(col =>
            {
                actual = actual.Replace($"({col})", $"(\"{col}\")");
            });
```

if index column has operator class, column wont be replaces and quotes will not be applied

as result 

ToDDL().EqualsIgnoreCase(actual);

will return false since

actual = `CREATE INDEX mt_doc_clip_idx_data ON public.mt_doc_clip USING gin (data jsonb_path_ops);`

ToDDL() = `CREATE INDEX mt_doc_clip_idx_data ON public.mt_doc_clip USING gin ("data" jsonb_path_ops);`

**Comment by jeremydmiller** [comment]
Cool, meant to tell you it had to be something we weren't doing right in comparing the actual to the expected. When Postgresql processes DDL, it frequently changes some details of it, so when you query the metadata, you don't necessarily get the same body text back and you end up having to massage both of the expected & actual before comparing. I clearly missed this one.

Do you have any interest in doing the PR to correct that by chance?

**Comment by vadimline** [comment]
Sure. How do you think would be better to fix this issue? Should I use Regex to replace column with column in quotes to ensure that correct part of script is replaced?

**Comment by jeremydmiller** [comment]
I don't have any better ideas. Sounds good to me.

**Comment by vadimline** [comment]
Ok, thanks, then I will provide PR soon.

**Comment by jeremydmiller** [comment]
Bueno, thank you.

**Comment by jeremydmiller** [comment]
Addressed by #704

**Comment by endofcake** [comment]
Thank you thank you thank you.
